### PR TITLE
Conditionally apply org id filter to project lookup

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Continues on https://github.com/gitpod-io/gitpod/pull/19217#discussion_r1420149024, which could potentially create some issues when looking up with a missing / empty organization ID.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-findpro638e2a9b67</li>
	<li><b>🔗 URL</b> - <a href="https://ft-findpro638e2a9b67.preview.gitpod-dev.com/workspaces" target="_blank">ft-findpro638e2a9b67.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-findprojsbycloneurl-review-comments-gha.21621</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-findpro638e2a9b67%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/typeorm/project-db-impl.ts [hunk 1]
```diff
@@ -62,7 +62,10 @@ export class ProjectDBImpl extends TransactionalDBImpl<ProjectDB> implements Pro
 
     public async findProjectsByCloneUrl(cloneUrl: string, organizationId?: string): Promise<Project[]> {
         const repo = await this.getRepo();
-        const conditions: FindConditions<DBProject> = { cloneUrl, markedDeleted: false, teamId: organizationId };
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/project-db-impl.ts:65]: :orange_heart: for adding a test for the function [here](https://github.com/gitpod-io/gitpod/blob/eaae6cc6f4f5ee1ec81db4c611502f5c75fca97d/components/gitpod-db/src/project-db.spec.db.ts#L169)!

Comment (filiptronicek) [reply components/gitpod-db/src/typeorm/project-db-impl.ts:65]: Working on it! 👨‍💻

Comment (filiptronicek) [reply components/gitpod-db/src/typeorm/project-db-impl.ts:65]: ✅

**Review by geropl — APPROVED** [review]
Code LGTM, and tests :orange_heart: are :green_circle: : :heavy_check_mark:

**Pull request merged**